lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3594,6 +3594,72 @@ class ConvertAtenIndexSelectOp : public OpConversionPattern<AtenIndexSelectOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    Location loc = op.getLoc();
+    MLIRContext *context = op->getContext();
+    Value input = adaptor.self();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    int64_t inputRank = inputType.getRank();
+    Type elementType = inputType.getElementType();
+
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    SmallVector<AffineExpr> indexExprs;
+    for (int i = 0; i < inputRank; i++) {
+      if (i == dim) {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()) -
+                             getAffineConstantExpr(1, context));
+      } else {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+      }
+    }
+    auto cumsumResultMap = AffineMap::get(inputRank, 0, indexExprs, context);
+    SmallVector<AffineMap> indexingMaps = {
+        rewriter.getMultiDimIdentityMap(inputRank), cumsumResultMap};
+    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
```

Comment:
nit: use `getParallelIteratorTypeName());`

Reply:
should it be "reduction" on the `dim` dimension?

Reply:
No, the input and output dimensions are same in cumsum. There is no reduction.